### ISSUE TITLE
[BUG] CMake Error at .../thrust-config-version.cmake

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids-jni#479]
**Describe the bug**
spark-rapids-jni_nightly-dev ID 184 failed at,
```
16:28:05  [INFO]      [exec] -- Found JNI: /usr/lib/jvm/java/jre/lib/amd64/libjawt.so  
16:28:05  [INFO]      [exec] -- JDK with JNI in /usr/lib/jvm/java/include;/usr/lib/jvm/java/include/linux;/usr/lib/jvm/java/include
16:28:05  [INFO]      [exec] -- Found nvcomp: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib/cmake/nvcomp/nvcomp-config.cmake (found version "2.3.3") 
16:28:05  [INFO]      [exec] -- Found Thrust: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/include/libcudf/Thrust/thrust/cmake/thrust-config.cmake (found version "1.15.0.0") 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:10 (file):
16:28:05  [INFO]      [exec]   file failed to open for reading (No such file or directory):
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec]     /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/src/main/cpp/_THRUST_VERSION_INCLUDE_DIR-NOTFOUND/thrust/version.h
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:30 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:17 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: " / 100000": syntax error, unexpected
16:28:05  [INFO]      [exec]   exp_DIVIDE (2).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:30 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:18 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: "( / 100) % 1000": syntax error,
16:28:05  [INFO]      [exec]   unexpected exp_DIVIDE (3).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:30 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:19 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: " % 100": syntax error, unexpected
16:28:05  [INFO]      [exec]   exp_MOD (2).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:30 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:10 (file):
16:28:05  [INFO]      [exec]   file failed to open for reading (No such file or directory):
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec]     /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/src/main/cpp/_THRUST_VERSION_INCLUDE_DIR-NOTFOUND/thrust/version.h
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:31 (find_dependency)
16:28:05  [INFO]      [exec] -- Found rmm: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake (found version "22.10.0")   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:17 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: " / 100000": syntax error, unexpected
16:28:05  [INFO]      [exec]   exp_DIVIDE (2).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:31 (find_dependency)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:18 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: "( / 100) % 1000": syntax error,
16:28:05  [INFO]      [exec]   unexpected exp_DIVIDE (3).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:31 (find_dependency)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/thrust/thrust-config-version.cmake:19 (math):
16:28:05  [INFO]      [exec]   math cannot parse the expression: " % 100": syntax error, unexpected
16:28:05  [INFO]      [exec]   exp_MOD (2).
16:28:05  [INFO]      [exec] Call Stack (most recent call first):
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-dependencies.cmake:24 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/rmm/rmm-config.cmake:75 (include)
16:28:05  [INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:31 (find_dependency)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
16:28:05  [INFO]      [exec]   /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:114 (find_package)
16:28:05  [INFO]      [exec]   CMakeLists.txt:104 (rapids_find_package)
16:28:05  [INFO]      [exec] 
16:28:05  [INFO]      [exec] 
16:28:06  [INFO]      [exec] -- Check if compiler accepts -pthread
16:28:06  [INFO]      [exec] -- Check if compiler accepts -pthread - yes
16:28:06  [INFO]      [exec] -- Found libcudacxx: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/libcudacxx/libcudacxx-config.cmake (found version "1.7.0") 
16:28:06  [INFO]      [exec] -- Found cuco: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cuco/cuco-config.cmake (found version "0.0.1") 
16:28:06  [INFO]      [exec] -- Found cuFile: /usr/local/cuda/lib64/libcufile.so  
16:28:06  [INFO]      [exec] -- Found KvikIO: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/kvikio/kvikio-config.cmake (found version "22.10.0") 
16:28:06  [INFO]      [exec] -- Found ZLIB: /usr/lib64/libz.so (found version "1.2.7") 
16:28:06  [INFO]      [exec] -- Found cudf: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake (found version "22.10.0") 
16:28:06  [INFO]      [exec] -- Found GTest: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/libcudf-install/lib64/cmake/GTest/GTestConfig.cmake (found version "1.10.0")  
16:28:06  [INFO]      [exec] -- Found Boost: /usr/local/lib/cmake/Boost-1.79.0/BoostConfig.cmake (found version "1.79.0")  
16:28:06  [INFO]      [exec] -- Configuring incomplete, errors occurred!
16:28:06  [INFO]      [exec] See also "/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/CMakeFiles/CMakeOutput.log".
16:28:06  [INFO]      [exec] See also "/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build/CMakeFiles/CMakeError.log".
16:28:06  [INFO] ------------------------------------------------------------------------
16:28:06  [INFO] BUILD FAILURE
16:28:06  [INFO] ------------------------------------------------------------------------
16:28:06  [INFO] Total time: 1:26:13.167s
16:28:06  [INFO] Finished at: Sat Aug 13 08:28:06 UTC 2022
16:28:07  [INFO] Final Memory: 19M/654M
16:28:07  [INFO] ------------------------------------------------------------------------
16:28:07  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (build-sparkrapidsjni) on project spark-rapids-jni: An Ant BuildException has occured: exec returned: 1
16:28:07  [ERROR] around Ant part ...<exec failonerror="true" dir="/home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/cmake-build" executable="cmake">... @ 5:152 in /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-184-cuda11/target/antrun/build-main.xml
16:28:07  [ERROR] -> [Help 1]
16:28:07  [ERROR] 
16:28:07  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
16:28:07  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
16:28:07  [ERROR] 
16:28:07  [ERROR] For more information about the errors and possible solutions, please read the following articles:
16:28:07  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by revans2** [comment]
Looking at this a little bit, because it is blocking everyone right now. I think something in rapids-cmake changed and because cudf does not pin this our builds are not reproducible.

https://github.com/rapidsai/rapids-cmake/commit/23095268134adcd7e6c093f54b8625b24a76bd0a

Will see if we can fix this.

**Comment by revans2** [comment]
It looks like there is no simple way to revert the changes in rapids-cmake. We would have to patch cudf and rapids-cmake to make this work. I think the fastest solution to the problem is to fix/work around https://github.com/rapidsai/cudf/issues/11521

**Comment by revans2** [comment]
I filed https://github.com/rapidsai/cudf/issues/11533 because our builds cannot be reproduced.

**Comment by jlowe** [comment]
This error should be fixed after rapidsai/cudf#11535 is merged and spark-rapids-jni is able to update to the latest cudf changes.
